lib/wave-1-scoring.rb

```diff
@@ -0,0 +1,64 @@
+module Scrabble
+  class Scoring
+    SCORE_HASH = {
+
+      %w(A E I O U L N R S T) => 1,
+      %w(D G) => 2,
+      %w(B C M P) => 3,
+      %w(F H V W Y) => 4,
+      %w(K) => 5,
+      %w(J X) => 8,
+      %w(Q Z) => 10
+    }  #letter_arr , letter_score
+
+    def self.score(word)
+      raise ArgumentError.new "Input must be a String class." if word.class != String
+      raise ArgumentError.new "Word must be one to seven letters long." if word.length == 0 || word.length > 7
+      raise ArgumentError.new "Word must contain only regular ASCII letters." if !word.match(/^[A-Z]+$/i)
+
+      letters = word.upcase.split("")
+      word_score = 0
+
+      letters.each do |letter|
+        SCORE_HASH.each do |letter_arr, letter_score|
+          if letter_arr.include? letter
+            word_score += letter_score
+          end
+        end
+      end
```

Comment:
this could probably be:

```
letters.each do |letter|
  SCORE_HASH.keys.each do |key|
    if key.include? letter
      word_score += SCORE_HASH[key]
    end
  end
end
```